core/src/executor/aggregate/mod.rs

```diff
@@ -156,25 +171,47 @@ impl<'a> Aggregator<'a> {
     }
 }
 
-fn aggregate<'a>(
+#[async_recursion(?Send)]
+async fn aggregate<'a>(
     state: State<'a>,
-    context: &BlendContext<'_>,
+    blend_context: Rc<BlendContext<'async_recursion>>,
+    filter_context: Option<Rc<FilterContext<'a>>>,
     expr: &'a Expr,
 ) -> Result<State<'a>> {
-    let aggr = |state, expr| aggregate(state, context, expr);
-
+    let aggr = |state, expr| {
+        aggregate(
+            state,
+            Rc::clone(&blend_context),
+            filter_context.clone(),
```

Comment:
It would be nice to explicitly write `Rc::clone`.
https://doc.rust-lang.org/stable/std/rc/struct.Rc.html#method.clone

```suggestion
            filter_context.as_ref().map(Rc::clone),
```